src/canister/individual_user_template/src/api/canister_management/migrate_hotornot_user_to_yral.rs

```diff
@@ -0,0 +1,158 @@
+
+    if profile_owner != caller() {
+        return Err("Unauthorized".to_owned());
+    }
+
+    // Should be in hotornot subnet
+    if check_canister_is_in_hotornot_subnet(profile_owner, true)
+        .await
+        .is_err()
+    {
+        return Err("Unauthorized".to_owned());
+    }
+
+    CANISTER_DATA.with_borrow_mut(|canister_data| {
+        let result = canister_data
+            .session_type
+            .ok_or(String::from("Canister not yet assigned"));
+        canister_data.last_access_time = Some(get_current_system_time_from_ic());
+        result
+    })?;
+
+    let current_time = get_current_system_time_from_ic();
+    let amount = CANISTER_DATA
+        .with_borrow(|canister_data| canister_data.my_token_balance.utility_token_balance);
+    let all_created_posts =
+        CANISTER_DATA.with_borrow(|canister_data| canister_data.all_created_posts.clone());
+
+    match ic_cdk::call::<(u64, Principal, BTreeMap<u64, Post>), (Result<bool, String>,)>(
+        to_account,
+        "receive_data_from_hotornot",
+        (amount, profile_owner, all_created_posts),
+    )
+    .await
+    {
+        Ok(_) => CANISTER_DATA.with_borrow_mut(|canister_data| {
+            canister_data
+                .my_token_balance
+                .handle_token_event(TokenEvent::Transfer {
+                    amount,
+                    to_account,
+                    timestamp: current_time,
+                });
+        }),
+        Err(error) => {
+            return Err(format!("{:?}: {}", error.0, error.1));
+        }
+    }
+
+    Ok("Success".into())
+}
+
+async fn check_canister_is_in_hotornot_subnet(
+    canister_id: Principal,
+    is_into_subnet: bool,
+) -> Result<bool, String> {
+    match canister_info(CanisterInfoRequest {
+        canister_id,
+        num_requested_changes: None,
+    })
+    .await
+    {
+        Ok(canister_response) => {
+            let mut matched = false;
+            for controller in canister_response.0.controllers {
+                if controller.to_text().eq(HOT_OR_NOT_CONTROLLER) {
+                    matched = true;
+                    break;
+                }
+            }
+            Ok(is_into_subnet == matched)
+        }
+        Err(error) => Err(format!("{:?} : {}", error.0, error.1)),
+    }
+}
+
+#[update]
+pub async fn receive_data_from_hotornot(
+    amount: u64,
+    from_account: Principal,
+    all_created_posts: BTreeMap<u64, Post>,
+) -> Result<String, String> {
+    if CANISTER_DATA.with_borrow(|canister_data| {
+        canister_data
+            .migration_status
+            .is_some_and(|v| v.eq(&MigrationStatus::MigratedToYral))
+    }) {
+        return Err("Already Migrated".to_owned());
+    };
+
+    let profile_owner =
+        CANISTER_DATA.with_borrow(|canister_data| canister_data.profile.principal_id.unwrap());
+
+    if profile_owner != caller() {
+        return Err("Unauthorized".to_owned());
+    }
```

Comment:
This wouldn't be called from profile owner. This will be called from hotornot canister profile owner. We can only check if that canister belongs to hotornot subnet.